packages/event-migration/src/read-models-migration-check.ts

```diff
@@ -100,30 +108,34 @@ function connectToReadModelDb({
   return client.db(database);
 }
 
-export async function compareReadModelsCollection(
+export async function compareReadModelsCollection<
+  TSchema extends z.ZodSchema<any>, // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
`any` here is needed because I want this generic to accept all the shape of zod schemas possible while still being able to use the data results. 
Using `unknown` would not have allowed me to pass the results inside the `zipDataById`.